pkg/data-source-validator/data_source_validator.go

```diff
@@ -91,13 +92,12 @@ func NewDataSourceValidator(
 		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "pvc"),
 	}
 
-	pvcInformer.Informer().AddEventHandlerWithResyncPeriod(
+	pvcInformer.Informer().AddEventHandler(
```

Comment:
@pohly This version doesn't perform as well as the earlier version with the resource check. My metrics counters continue to tick up over time even though I'm not creating new volumes.

Reply:
Okay I added the resource version check back in and it no longer reprocesses objects.

You're right that the resync was unneeded, but for some reason without the resource version check we're still seeing the same objects over and over.

Reply:
It would be good to get to the bottom of this. Perhaps ask on #sig-apimachinery?

Reply:
@pohly I noticed while removing the resyncPeriod that it's still used when constructing the informers. See:

https://github.com/kubernetes-csi/volume-data-source-validator/blob/master/cmd/volume-data-source-validator/main.go#L97

The shared informer constructor requires a resync period. It's a non-optional field.

Reply:
Okay I've done substantially more searching on this subject. It seems that you can pass zero as the resyncPeriod and that disables resyncs altogether. It's very poorly documented, but this is supposed to be safe if your controller doesn't interact with anything external. I will try this.